workflow-controller/rockcraft.yaml

```diff
@@ -49,43 +50,36 @@ parts:
       - libcap2-bin
       - zip
     override-build: |
-      # builder stage https://github.com/argoproj/argo-workflows/blob/v3.3.8/Dockerfile#L36 
+      # builder stage
+      # https://github.com/argoproj/argo-workflows/blob/v3.4.17/Dockerfile#L21
       go mod download
 
-      # workflow-controller-build stage https://github.com/argoproj/argo-workflows/blob/v3.3.8/Dockerfile#L94
-      cat .dockerignore >> .gitignore
-      git status --porcelain | cut -c4- | xargs git update-index --skip-worktree
+      # reset LDFLAGS because Go has different format, i.e. '-L' is undefined in Go
+      LDFLAGS=""
+
+      # workflow-controller-build
+      # https://github.com/argoproj/argo-workflows/blob/v3.4.17/Dockerfile#L56
       make dist/workflow-controller
 
-      # workflow-controller stage https://github.com/argoproj/argo-workflows/blob/v3.3.8/Dockerfile#L131
-      ls -la /usr/share/
+      # workflow-controller stage
+      # https://github.com/argoproj/argo-workflows/blob/v3.4.17/Dockerfile#L90
       install -DTd "/usr/share/zoneinfo" "$CRAFT_PART_INSTALL/usr/share/zoneinfo"
-      chown 8737 $CRAFT_PART_INSTALL/usr/share/zoneinfo
       install -DT "./dist/workflow-controller" "$CRAFT_PART_INSTALL/bin/workflow-controller"
-      chown 8737 $CRAFT_PART_INSTALL/bin/workflow-controller
 
   copy-files:
     plugin: dump
     after: [builder]
     source: https://github.com/argoproj/argo-workflows
     source-type: git
     source-subdir: hack
-    source-tag: v3.3.8
+    source-tag: v3.4.17
     organize:
       ssh_known_hosts: etc/ssh/ssh_known_hosts
       nsswitch.conf: etc/ssh/nsswitch.conf
+      arch.sh: bin/arch.sh
+      os.sh: bin/os.sh
     stage:
       - etc/ssh/ssh_known_hosts
       - etc/ssh/nsswitch.conf
-
-  # using user 8737 because it was used in workload Dockerfile
-  non-root-user:
-    plugin: nil
-    after: [copy-files]
-    overlay-script: |
-      # Create a user in the $CRAFT_OVERLAY chroot
-      mkdir -p $CRAFT_OVERLAY/etc
-      chmod 755 $CRAFT_OVERLAY/etc
-      groupadd -R $CRAFT_OVERLAY -g 8737 ubuntu
-      useradd -R $CRAFT_OVERLAY -M -r -u 8737 -g ubuntu ubuntu
-
+      - bin/arch.sh
+      - bin/os.sh
```

Comment:
Why do we need the added ones? I don't see them being referenced in the upstream dockerfile.